misc/server/Dockerfile

```
@@ -1,8 +1,12 @@
-FROM rust:1.66-bullseye as builder
+FROM rust:1.67-bullseye as builder
 WORKDIR /usr/src/rust-libp2p-server
 
 # Run with access to the target cache to speed up builds
 WORKDIR /workspace
+
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    cargo install --root /usr/local libp2p-lookup
```

```suggestion
    cargo install --root /usr/local libp2p-lookup  --version 0.6.4
```

To make this as deterministic as possible.